### ISSUE TITLE
Update aws_eks_node_group to allow a min_size and desired_size of zero

### ISSUE THREAD

**Issue by Raminios** [hashicorp/terraform-provider-aws#20116]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

This is effectively a re-opening of issue [#13984](https://github.com/hashicorp/terraform-provider-aws/issues/13984), as it has been closed to comments.

The [response](https://github.com/hashicorp/terraform-provider-aws/issues/13984#issuecomment-651425676) to the original issue links to an [upstream ticket](https://github.com/aws/containers-roadmap/issues/724) on the AWS side referencing that their API does not currently support zero values, and the module is just applying that same validation.

The API has been updated with [version 3.46.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.46.0), and the [documentation](https://docs.aws.amazon.com/eks/latest/APIReference/API_NodegroupScalingConfig.html#AmazonEKS-Type-NodegroupScalingConfig-minSize) linked in the original ticket now shows the following:

> desiredSize
The current number of nodes that the managed node group should maintain.
Type: Integer
Valid Range: Minimum value of 0.

>minSize
The minimum number of nodes that the managed node group can scale in to.
Type: Integer
Valid Range: Minimum value of 0.

The upstream ticket remains open for the reason listed [here](https://github.com/aws/containers-roadmap/issues/724#issuecomment-865187007), but it feels like the limitations within the module can now be lifted so that `0` can be provided for scaling configs?

### New or Affected Resource(s)

*aws_eks_node_group

### Potential Terraform Configuration

```hcl
resource "aws_eks_node_group" "eks_node_group" {
  cluster_name    = locals.eks_cluster_name
  node_group_name = locals.eks_node_group_name
  node_role_arn   = locals.eks_worker_role_arn
  instance_types  = [locals.eks_instance_type]
  subnet_ids      = [locals.eks_subnet_id]

  scaling_config {
    desired_size = 0
    min_size = 0
    max_size = 50
  }

  lifecycle {
    ignore_changes = [
      scaling_config[0].desired_size,
    ]
  }
}
```

### References

* https://github.com/hashicorp/terraform-provider-aws/issues/13984
* https://github.com/aws/containers-roadmap/issues/724
* https://github.com/hashicorp/terraform-provider-aws/releases/tag/v3.46.0
* https://docs.aws.amazon.com/eks/latest/APIReference/API_NodegroupScalingConfig.html#AmazonEKS-Type-NodegroupScalingConfig-minSize

**Comment by daenney** [comment]
This was fixed in #19810 I believe.

**Comment by anton-mesnyankin** [comment]
I'm able to repro the issue with hashicorp/aws provider v4.31.0

my config is attached [eks_group_issue.txt](https://github.com/hashicorp/terraform-provider-aws/files/9614737/eks_group_issue.txt) due to formating issue.

output is:

> Error: expected scaling_config.0.max_

size to be at least (1), got 0
> 
>   with aws_eks_node_group.backend-1c-g31,
>   on backend.tf line 146, in resource "aws_eks_node_group" "backend-1c-g31":
>  146:     max_size     = 0

**Comment by mattburgess** [comment]
@anton-mesnyankin that error is for the `max_size` setting not `min_size` which this ticket is about. The [API](https://docs.aws.amazon.com/sdk-for-go/api/service/eks/#NodegroupScalingConfig) prevents setting that to below 1, so we enforce the same restriction in the provider as well.

Closing as this was fixed in [v3.46.0 of the Terraform AWS Provider](https://github.com/hashicorp/terraform-provider-aws/blob/v3.46.0/CHANGELOG.md).
